applications/FluidDynamicsApplication/custom_elements/embedded_fluid_element_discontinuous.cpp

```diff
@@ -928,17 +951,36 @@ void EmbeddedFluidElementDiscontinuous<TBaseElement>::CalculateDragForce(
             // Get the interface Gauss pt. unit noromal
             const auto &aux_unit_normal = rData.NegativeInterfaceUnitNormals[g];
 
-            // Compute Gauss pt. pressure
+            // Compute Gauss pt. values
             const double p_gauss = inner_prod(rData.N, rData.Pressure);
+            const array_1d<double, Dim> v_gauss = prod(rData.N, rData.Velocity);
+            array_1d<double,Dim> v_emb_gauss = ZeroVector(Dim);
+            for (unsigned int i_node = 0; i_node < NumNodes; ++i_node) {
+                const auto &r_i_emb_vel = r_geom[i_node].GetValue(EMBEDDED_VELOCITY);
+                for (unsigned int d = 0; d < Dim; ++d) {
+                    v_emb_gauss(d) += r_i_emb_vel(d) * rData.N(i_node);
+                }
+            }
 
             // Get the normal projection matrix in Voigt notation
             BoundedMatrix<double, Dim, StrainSize> voigt_normal_proj_matrix = ZeroMatrix(Dim, StrainSize);
             FluidElementUtilities<NumNodes>::VoigtTransformForProduct(aux_unit_normal, voigt_normal_proj_matrix);
+            BoundedMatrix<double, Dim, Dim> norm_proj_matrix, tang_proj_matrix;
+            FluidElementUtilities<NumNodes>::SetNormalProjectionMatrix(aux_unit_normal, norm_proj_matrix);
+            FluidElementUtilities<NumNodes>::SetTangentialProjectionMatrix(aux_unit_normal, tang_proj_matrix);
 
             // Add the shear and pressure drag contributions
             const array_1d<double, Dim> shear_proj = rData.Weight * prod(voigt_normal_proj_matrix, rData.ShearStress);
```

Comment:
For understanding:
What does the voigt_normal_proj_matrix do? A projection onto the interface surface? And then what does the norm_proj_matrix and tang_proj_matrix do afterwards? Do they give the normal and tangential part of the shear stress tangential to the surface of the interface? Why is the tangential part not taken from the shear force, but calculated using the velocity?

Reply:
Okay, I think I understood that last question and for the remaining questions I will assume "yes" until contradiction ;)

Reply:
`voigt_normal_proj_matrix` normal projection matrix for a 2nd order tensor in Voigt Notation
`norm_proj_matrix` 2nd order normal projection operator (n x n)
`tang_proj_matrix` 2nd order tangential projection operator (I - n x n)

What we do in here is to decompose the viscous component of the drag force in a normal and tangential part using such operators. Then, the tangential part is calculated from the Navier-slip BC in order to avoid as much as possible using the Ausas FE space gradients. This is developed in the 5th chapter of my thesis.

Reply:
Thanks!